frontend/src/data_context_global.tsx

```diff
@@ -210,17 +210,74 @@ export function asLocale(dateString?: string) {
  * @param genebank the genebank data to filter active individuals from
  * @param sex the sex of the active individuals
  */
-export function activeIndividuals(genebank: Genebank | undefined, sex: string) {
-  return React.useMemo(() => {
-    if (!genebank) {
-      return [];
-    }
-    return genebank?.individuals.filter(
-      (i) => i.sex == sex && i.is_active == true
-    );
-  }, [genebank]);
+export function activeIndividuals(
+  genebank: Genebank | undefined,
+  sex: string,
+  herdId: string | undefined
+) {
+  const origin: HerdNameID = {
+    herd: herdId,
+  };
+  if (!genebank) {
+    return [];
+  }
+  return genebank?.individuals.filter(
+    (i) =>
+      i.sex == sex &&
+      i.is_active == true &&
+      (herdId ? i.herd.herd == origin.herd : true)
+  );
+}
+
+/**
+ * Returns alive individuals of given sex in the genebank from a given date
+ * @param genebank the genebank data to filter active individuals from
+ * @param sex the sex of the active individuals
+ * @param fromDate the latest birth date from which individuals should be retrieved
+ */
```

Comment:
Ahh, really good with a proper comment. :D I should become better on that as well...